homeassistant/components/surepetcare/__init__.py

```diff
@@ -0,0 +1,144 @@
+"""Support for Sure Petcare cat/pet flaps."""
+import logging
+
+import voluptuous as vol
+from homeassistant.const import (CONF_DEVICE_CLASS, CONF_ICON, CONF_ID,
+                                 CONF_NAME, CONF_PASSWORD, CONF_SCAN_INTERVAL,
+                                 CONF_TYPE, CONF_USERNAME)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import (CONF_FLAPS, CONF_HOUSEHOLD_ID, CONF_PETS,
+                    DATA_SURE_PETCARE, DATA_SUREPY, DEFAULT_SCAN_INTERVAL,
+                    DOMAIN, SURE_IDS, TOPIC_UPDATE, SureThingID)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SCAN_INTERVAL = DEFAULT_SCAN_INTERVAL
+
+FLAP_SCHEMA = vol.Schema({
+    vol.Required(CONF_ID): cv.positive_int,
+    vol.Required(CONF_NAME): cv.string,
+})
+
+PET_SCHEMA = vol.Schema({
+    vol.Required(CONF_ID): cv.positive_int,
+    vol.Required(CONF_NAME): cv.string,
+})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_HOUSEHOLD_ID): cv.positive_int,
+        vol.Required(CONF_FLAPS): vol.Schema(
+            vol.All(cv.ensure_list, [FLAP_SCHEMA])),
+        vol.Required(CONF_PETS): vol.Schema(
+            vol.All(cv.ensure_list, [PET_SCHEMA])),
+        vol.Optional(CONF_DEVICE_CLASS, default="door"): cv.string,
+        vol.Optional(CONF_ICON, default="mdi:door"): cv.string,
+        vol.Optional(
+            CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, config):
+    """Initialize the Sure Petcare component."""
+    from surepy import SurePetcare
+
+    # config file data
+    conf = config.get(DOMAIN, dict())
+
+    # shared data
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][CONF_SCAN_INTERVAL] = conf[CONF_SCAN_INTERVAL]
+    hass.data[DATA_SURE_PETCARE] = {}
+    hass.data[DATA_SURE_PETCARE][CONF_USERNAME] = CONF_USERNAME
+    hass.data[DATA_SURE_PETCARE][CONF_PASSWORD] = CONF_PASSWORD
+    hass.data[DATA_SURE_PETCARE][CONF_HOUSEHOLD_ID] = conf[CONF_HOUSEHOLD_ID]
+    hass.data[DATA_SURE_PETCARE][SureThingID.FLAP.name] = dict()
+    hass.data[DATA_SURE_PETCARE][SureThingID.PET.name] = dict()
+
+    # sure petcare api connection
+    hass.data[DATA_SURE_PETCARE][DATA_SUREPY] = SurePetcare(
```

Comment:
We should test that the credentials are correct before continuing with the setup. If they are incorrect we should log an error and `return False`.